### PR TITLE
Handle `decrease`/`secrets` conflicts with `ECR` cases to initialism

### PR DESCRIPTION
**Pull request by Julian-Chu**
Description of changes:

prevent names contain `decrease` to be generated for ECR case.

example:  two  DDB data fields 
- `LastDecreaseDateTime`
- `NumberOfDecreasesToday`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: names/names_test.go [hunk 1]
```diff
@@ -105,6 +105,8 @@ func TestNames(t *testing.T) {
 		{"Throttling", "Throttling", "throttling", "throttling", "throttling"},
 		{"UUID", "UUID", "uuid", "uuid", "uuid"},
 		{"Vlan", "VLAN", "vlan", "vlan", "vlan"},
+		{"LastDecreaseDateTime", "LastDecreaseDateTime", "lastDecreaseDateTime", "last_decrease_date_time", "lastdecreasedatetime"},
+		{"NumberOfDecreasesToday", "NumberOfDecreasesToday", "numberOfDecreasesToday", "number_of_decreases_today", "numberofdecreasestoday"},
```
Comment (a-hilaly) [inline names/names_test.go:109]: Ideally we need to also add unit test case for `Secret` and `ECR`

File: names/names.go [hunk 2]
```diff
@@ -110,7 +110,8 @@ var (
 		{"Ebs", "EBS", "ebs", nil},
 		{"Ec2", "EC2", "ec2", nil},
 		// Prevent "Secret" from becoming "s_ecr_et"
-		{"Ecr", "ECR", "ecr", re2.MustCompile("(?!S|s)ecr(?!et)", re2.None)},
+		// Prevent "Decrease" from becoming "d_ecr_ease"
+		{"Ecr", "ECR", "ecr", re2.MustCompile("(?!S|s|D|d)ecr(?!et|ease)", re2.None)},
```
Comment (a-hilaly) [inline names/names.go:114]: The test case `{"Ecr", "ECR", "ecr", "ecr", "ecr"},` does not pass with this regex

**Review by a-hilaly — COMMENTED** [review]
Thank you @Julian-Chu !
Have two comments below
/ok-to-test

File: names/names_test.go (hunk 1 above)
Comment (Julian-Chu) [reply names/names_test.go:110]: - add `ECR` test case, `Secret` exists.

File: names/names.go (hunk 2 above)
Comment (Julian-Chu) [reply names/names.go]: apply new regex, but just do try and error 😓 ,   if you have idea to improve the regex,  please let me know.

File: names/names_test.go [hunk 3]
```diff
@@ -94,7 +95,7 @@ func TestNames(t *testing.T) {
 		{"RepositoryUriTest", "RepositoryURITest", "repositoryURITest", "repository_uri_test", "repositoryuritest"},
 		{"RequestedAmiVersion", "RequestedAMIVersion", "requestedAMIVersion", "requested_ami_version", "requestedamiversion"},
 		{"SaslScram512Auth", "SASLSCRAM512Auth", "saslSCRAM512Auth", "sasl_scram_512_auth", "saslscram512auth"},
-		{"Secret", "Secret", "secret", "secret", "secret"},
+		{"Secret", "SECRET", "secret", "secret", "secret"},
```
Comment (a-hilaly) [inline names/names_test.go:98]: this should be: `{"Secret", "Secret", "secret", "secret", "secret"}`

**Review by a-hilaly — CHANGES_REQUESTED** [review]

File: names/names_test.go (hunk 3 above)
Comment (Julian-Chu) [reply names/names_test.go]: fixed [- remove Secret from initialisms](https://github.com/aws-controllers-k8s/pkg/pull/21/commits/b2030dde27058a276fc32dd5c5f8db7cff3e36c7)

File: names/names.go (hunk 2 above)
Comment (a-hilaly) [reply names/names.go]: Shoot.. i just saw this. Glad you figured it out!

**Review by a-hilaly — APPROVED** [review]
Thank you sir!
/lgtm

**Pull request merged**